packit/api.py

```diff
@@ -826,12 +824,48 @@ def get_upstream_release_monitoring_bug(
 
         return None
 
+    def determine_version(self, versions: list[str], branch: str):
+        """
+        Determine version to propose the update for considering
+        the version in the specfile of the particular branch
+        and version_update_mask configuration option.
+        """
+        logger.info(f"Determining version to propose from {versions}")
+        downstream_spec_ver = None
+
+        try:
+            downstream_spec_ver = self.dg.get_specfile_version()
+            self.dg.refresh_specfile()
+            logger.debug(f"Version in specfile: {downstream_spec_ver}")
+        except Exception as ex:
+            logger.debug(ex)
+
+        if not downstream_spec_ver:
+            logger.debug(
+                "Specfile version not found, picking the first one from the list.",
+            )
+            return versions[0]
+
+        # should be ordered from the latest
+        for version_ in versions:
+            # we could check here also upstream_tag_include and upstream_tag_exclude
+            if self.check_version_distance(
+                downstream_spec_ver,
+                version_,
+                branch,
+            ):
+                logger.info(f"Determined version: {version_}")
+                return version_
+
+        return None
+
     @overload
     def sync_release(
         self,
         dist_git_branch: Optional[str] = None,
         version: Optional[str] = None,
         tag: Optional[str] = None,
+        versions: Optional[list[str]] = None,
```

Comment:
The initial idea I had, was to remove (or at least deprecate) the `version` and use just `versions`, since it doesn't really matter whether there is just one version in the list or multiple :thinking: 

Now that I'm thinking about it, we could even have version as a local variable, start by populating it (from nothing, the `if`s you have expanded to check for `versions`), and then just use as we did before.

Reply:
I wanted to be rather explicit here whether there is one concrete version to propose or it should be determined in the method later (this cannot happen right in the beginning since we use the checked-out branch to get the current version in specfile for comparisons); so that way it seemed to me like less intrusive

also, in packit-service, we currently utilise the `tag` rather than `version`, see [here](https://github.com/packit/packit-service/blob/d95578f528b9aa8c8d40c97c87dae66ea92276e0/packit_service/worker/handlers/distgit.py#L246), so the code for anitya will need to be still changed

but I am not strongly against, so I can rework it that way if you prefer that solution

Reply:
reworked, PTAL :pray: